components/server/src/database/measurements.py

```diff
@@ -72,86 +74,16 @@ def update_measurement_end(database: Database, measurement_id: MeasurementId):
     return database.measurements.update_one(filter={"_id": measurement_id}, update={"$set": {"end": iso_timestamp()}})
 
 
-def insert_new_measurement(
-    database: Database, data_model, metric_data: dict, measurement: dict, previous_measurement: dict
-) -> dict:
+def insert_new_measurement(database: Database, measurement: Measurement) -> Measurement:
```

Comment:
This could be a class method on the Measurement class:

Measurement.insert(metric: Metric, measurement_data: dict)

Reply:
Same comment as above: looks nice, but mixes model and database layer. Let's discuss.